File: app/src/main/java/com/arcao/wherigoservice/api/WherigoApiFactory.java
package com.arcao.wherigoservice.api;

import com.arcao.geocaching.api.GeocachingApiFactory;

public final class WherigoApiFactory {
    public static WherigoService create() {
        return new WherigoServiceImpl(GeocachingApiFactory.getJsonDownloader());
    }
}

File: app/src/main/java/com/arcao/wherigoservice/api/WherigoApiFactory.kt
package com.arcao.wherigoservice.api

import org.koin.standalone.KoinComponent
import org.koin.standalone.get

@Deprecated("Use Koin")
object WherigoApiFactory : KoinComponent {
    @JvmStatic
    fun create() = get<WherigoService>()
}
